backend/announcements/serializers.py

```diff
@@ -0,0 +1,57 @@
+from announcements.models import Announcement, Audience
+from rest_framework import serializers
+
+
+class AudienceSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Audience
+        fields = ("name",)
+
+
+class AnnouncementSerializer(serializers.ModelSerializer):
+    audiences = serializers.SlugRelatedField(
+        many=True, slug_field="name", queryset=Audience.objects.all()
+    )
+
+    class Meta:
+        model = Announcement
+        fields = (
+            "id",
+            "title",
+            "message",
+            "announcement_type",
+            "release_time",
+            "end_time",
+            "audiences",
+        )
+
+    def to_representation(self, instance):
+        representation = super().to_representation(instance)
+        representation["audiences"] = [
+            audience.name for audience in instance.audiences.all()
```

Comment:
I think this might also be an `N+1` query here, might need to prefetch the `name` field for the audience

Reply:
Yep apologies, misread on my side. This is not a `N+1` query

Reply:
Also for reference, I don't think it is needed here unless you want to, but for future situations you can also do something like:
`names = instance.audiences.values_list('name', flat=True)`
to avoid the list comprehension